lfshttp/endpoint.go

```diff
@@ -119,6 +117,17 @@ func EndpointFromLocalPath(path string) Endpoint {
 		}
 		path = abs
 	}
+
+	var gitpath string
+	if filepath.Base(path) == ".git" {
+		gitpath = path
+		path = filepath.Dir(path)
+	} else {
+		gitpath = filepath.Join(path, ".git")
+	}
+	if file, err := os.Lstat(gitpath); err == nil && file.IsDir() {
```

Comment:
It's possible to have a file as the `.git` path (referred to as a _gitlink_), such as in a submodule or worktree, and I think we'd definitely want to let people push into submodules.  This is a tricky corner case, but I think we'd probably be fine with just checking if it exists.

It also occurred to me that this behavior is broken in our standalone helper, so I'll fix that in an independent PR.

Reply:
Actually, this is not broken in our standalone helper (:tada:), but we lack tests for it, so I'm going to add those tests in a separate PR.

Reply:
OK, I've adjusted this to just use `os.Stat()` and not try to check that the `.git` segment refers to a directory, just that it exists.